src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets

```diff
@@ -169,6 +171,14 @@ The .NET Foundation licenses this file to you under the MIT license.
       <NativeLibrary Condition="'$(UseSystemZlib)' != 'true'" Include="$(IlcSdkPath)libz.a" />
     </ItemGroup>
 
+    <!-- brotli must be added after System.IO.Compression.Native and brotlicommon must be added last, order matters. -->
+    <ItemGroup Condition="'$(UseSystemBrotli)' != 'true'">
+      <NativeLibrary Include="$(IlcFrameworkNativePath)libbrotlienc.a" />
```

Comment:
We get libz from IlcSdkPath. What's the reason for brotli to be different?

Reply:
Getting `libz` from the IlcSdkPath works well because we end up building zlib with the Mono runtime and we build the zlib libs as part of the NativeAOT runtime build (that builds the libraries static libs).

For brotli, Mono doesn't currently build brotli in any capacity or link it into their runtime at all. As a result, putting the brotli libs into the ILC SDK and not into the "libraries native components static libs" folder causes Mono-based static linking scenarios to fail as they don't have anywhere to get the brotli symbols from. In particular, we hit this on iOS, a platform where we decided to continue using the system zlib implementation due to size constraints and linking considerations.

Reply:
The brotli libs are still under `IlcSdkPath` on Windows. Is that right?

It does not look right to me that configuration of Mono targets is forcing a directory structure of NAOT packages.

Reply:
The brotli libs are next to the Compression.Native libs for all platforms. On Windows this is IlcSdkPath, on non-Windows it is not.

Reply:
Should we have libz next to Compression.Native too to have some sort of consistency?

Reply:
I can try giving that a shot now that we aren't going to use a vendored zlib on mobile platforms.

Reply:
Trying in https://github.com/dotnet/runtime/pull/110378